pumps/mongo_aggregate.go

```diff
@@ -290,28 +275,30 @@ func (m *MongoAggregatePump) WriteData(ctx context.Context, data []interface{})
 			_, avgErr := analyticsCollection.Find(query).Apply(avgChange, &withTimeUpdate)
 
 			if m.dbConf.ThresholdLenTagList != -1 && (len(withTimeUpdate.Tags) > m.dbConf.ThresholdLenTagList) {
-				printAlert(withTimeUpdate, m.dbConf.ThresholdLenTagList)
+				m.printAlert(withTimeUpdate, m.dbConf.ThresholdLenTagList)
 			}
 
 			if avgErr != nil {
-				log.WithFields(logrus.Fields{
-					"prefix": analytics.MongoAggregatePrefix,
-				}).Error("AvgUpdate Failure: ", avgErr)
+				m.log.Error("AvgUpdate Failure: ", avgErr)
 				return m.HandleWriteErr(avgErr)
 			}
 
+			m.log.WithFields(logrus.Fields{
+				"collection": collectionName,
+			}).Debug("Wrote aggregated data for ", len(data), " records")
+
 			if m.dbConf.UseMixedCollection {
 				thisData := analytics.AnalyticsRecordAggregate{}
 				err := analyticsCollection.Find(query).One(&thisData)
 				if err != nil {
-					log.Error("Couldn't find query doc!")
+					m.log.Error("Couldn't find query doc!")
```

Comment:
Can we please print the err value with the query and this data

Reply:
@letzya `thisData` here is just an empty aggregated struct. If err != nil will imply that `thisData` is still empty.
Added the query field.